packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/model/clientmodel/ClientMethod.java

```diff
@@ -511,6 +511,16 @@ public void addImportsTo(Set<String> imports, boolean includeImplementationImpor
                 ClassType.BINARY_DATA.addImportsTo(imports, false);
                 ClassType.HTTP_REQUEST.addImportsTo(imports, false);
             }
+
+            // sync-stack, lro (+ pageable)
+            boolean isLroPageable = (type == ClientMethodType.PagingSyncSinglePage
+                && proxyMethod != null
+                && GenericType.Response(ClassType.BINARY_DATA).equals(proxyMethod.getReturnType().getClientType()));
+            if (settings.isSyncStackEnabled() && settings.isFluent()) {
+                if (type == ClientMethodType.LongRunningBeginSync || isLroPageable) {
+                    ClassType.SYNC_POLLER_FACTORY.addImportsTo(imports, false);
```

Comment:
again trying to learn - If the operation is both LRO and Pagable,  is it that the `SyncPoller` returns `PagedIterable<T>` as final result once the LRO-Polling completes? or, is it that each poll can return `List<T>>`?

Reply:
LRO + pageable is a rarely used pattern in management-plane.

From my memory, I think the pageable part is mostly cosmetic. The API is an LRO that returns
```
{
  "value": [ ... ]
}
```
but service want the final result of the LRO looks like a pageable (i.e. `PagedIterable<T>`), hence they mark this API also as pageable.

Therefore, AFAIK, there will not be a "nextLink" and a 2nd page call.

Reply:
thanks, got it, so SDK don't break in case service really start to return multiple pages.